src/core/internal/convert.d

```diff
@@ -633,3 +633,41 @@ const(ubyte)[] toUbyte(T)(ref T val) if (is(T == struct) || is(T == union))
         return (cast(const(ubyte)*)&val)[0 .. T.sizeof];
     }
 }
+
+char[] itoa(ulong n) pure nothrow @safe
```

Comment:
> It is only used in CTFE, so I didn't see the GC-centric interface as an issue.

Agreed. I was just pointing out a more general design issue, where an interface that returns mutable can't exploit immutability to return aliased results. As a random thought outside of this PR.

> If core.demangle and object were to use the same implementation, would core.internal.convert still be a good place for it?

I have am not very savvy wrt how druntime is organized. @MartinNowak should be able to advise.

If the interfaces don't allow consolidation, or it is two difficult, then I'm fine with it. I *would* at the very least prefer that they have similar implementations.